pilot/pkg/xds/discovery.go

```diff
@@ -543,10 +518,6 @@ func (s *DiscoveryServer) initPushContext(req *model.PushRequest, oldPushContext
 		return nil, err
 	}
 
-	if err := s.UpdateServiceShards(push); err != nil {
```

Comment:
Why remove this and `getNonK8sRegistries`. 

It was originally intended for ADSC

Reply:
Oh `MCP` registry was removed, that was long ago

Reply:
It was only used for the mock client in tests currently